### PR TITLE
Fix NamespaceClient.GetNamespaceInfoAsync()

### LINKED ISSUE #637: Management client doesn't return correct NamespaceInfo
Two issues:

1. `ManagementClient.GetNamespaceInfoAsync()` invocation for premium tier never returns number of MUs.
2. `NamespaceType` is always `NamespaceType.Others` as broker returns `Messaging` for ASB namespaces, not `ServiceBus`.

This is a broken functionality as it doesn't do what it supposed.

Tested version: 3.3.0

Running simple test on standard and premium namespaces returns the following:

image

### PR DESCRIPTION
**Pull request by SeanFeldman**
Fixes #637

### CODE REVIEW

**Comment by SeanFeldman** [comment]
This will need to be a **minor** release as we're adding to the public API of an existing type `NamespaceType`.

**Comment by SeanFeldman** [comment]
@nemakam should we deprecate `NamespaceType.ServiceBus` or leave it as-is?

File: test/Microsoft.Azure.ServiceBus.UnitTests/API/ApiApprovals.ApproveAzureServiceBus.approved.txt [hunk 1]
```diff
@@ -730,6 +730,7 @@ namespace Microsoft.Azure.ServiceBus.Management
     public enum NamespaceType
     {
         ServiceBus = 0,
+        Messaging = 1,
```
Comment (nemakam) [inline test/Microsoft.Azure.ServiceBus.UnitTests/API/ApiApprovals.ApproveAzureServiceBus.approved.txt]: This is actually not true. Problem is that `ServiceBus` is called `Messaging` in service side. Didnt realize when I renamed the enum. The correct thing is to rename `ServiceBus` to `Messaging`. But that is going to break things. So the alternative that I can think of is to manually handle the deserialization and if it is Messaging, then convert it to ServiceBus. :(

**Comment by SeanFeldman** [comment]
I can live with that, but we'll need to change that in v4.

Sean Feldman

**Comment by SeanFeldman** [comment]
@nemakam updated

**Comment by SeanFeldman** [comment]
@nemakam could you please raise an issue for 4.0.0 milestone to address https://github.com/Azure/azure-service-bus-dotnet/pull/638/commits/f10cae959227ab8d28e9fe5877b968b789a1df0e#diff-4f0427b57425e52bf6ea1d62facac6b9R69? Thanks.

**Comment by SeanFeldman** [comment]
@nemakam please review

**Comment by SeanFeldman** [comment]
@nemakam who could review this?

**Comment by nemakam** [comment]
I guess we could just leave it as-is.

---
In reply to: [457061141](https://github.com/Azure/azure-service-bus-dotnet/pull/638#issuecomment-457061141) [](ancestors = 457061141)

**Pull request merged**